internal/server/data/data.go

```diff
@@ -362,6 +362,9 @@ func list[T models.Modelable](tx GormTxn, p *Pagination, selectors ...SelectorFu
 func save[T models.Modelable](tx GormTxn, model *T) error {
 	db := tx.GormDB()
 	setOrg(tx, model)
+	if isOrgMember(model) {
+		db = ByOrgID(tx.OrganizationID())(db)
+	}
```

Comment:
I think this is the bit we were missing. `setOrg` makes it look like we're doing the right thing, but that only sets the org_id of the new row, it doesn't force the query to only operate on an ID of the correct org.

I looked back to see if this was accidentally removed at some point, but I didn't find it, so I think it was something we missed while adding org support.

Reply:
right. setting the org is enough for inserts, but not for updates. As is it would effectively allow you to steal records from another org to your org.